server/channels/app/expirynotify.go

```diff
@@ -47,11 +47,15 @@ func (a *App) NotifySessionsExpired() error {
 
 		errPush := a.sendToPushProxy(tmpMessage, session)
 		if errPush != nil {
-			a.CountNotificationReason(model.NotificationStatusError, model.NotificationTypePush, model.NotificationReasonPushProxySendError, tmpMessage.Platform)
+			reason := model.NotificationReasonPushProxySendError
+			if errPush.Error() == notificationErrorRemoveDevice {
+				reason = model.NotificationReasonPushProxyRemoveDevice
+			}
+			a.CountNotificationReason(model.NotificationStatusError, model.NotificationTypePush, reason, tmpMessage.Platform)
```

Comment:
How many error types does the push proxy could return? Wondering if we just have `reason` contain the `errPush.Error()` for all cases

Reply:
There are many errors that are probably never going to happen, like errors converting the message to json, or decoding the response.

Then any error that making the request may throw, which we have no control over.

Then are the errors sent back by the push proxy, which can be also many, from failed to read the message body. Also, many errors have variable information, like the when it does not recognize the version in the device id, which says...
`"Could not determine the app version in %v appVersion=%v"`

You can take a look to the `NewErrorPushResponse` references in the push proxy code.

So... adding the error without any control may have bad implications for the cardinality of the data.